packages/alfa-rules/src/sia-r66/rule.ts

```diff
@@ -35,31 +36,66 @@ export default Rule.Atomic.of<Page, Text, Question.Metadata>({
   evaluate({ device, document }) {
     return {
       applicability() {
-        return visit(document);
+        let disabledWidgetNames: Sequence<string> = Sequence.empty();
+        let textNodes: Sequence<Text> = Sequence.empty();
 
-        function* visit(node: Node): Iterable<Text> {
+        gather(document);
+        return getApplicableLinks();
+
+        function gather(node: Node): void {
+          // Gather all applicable text nodes
+          if (
+            node.parent().isSome() &&
+            test(
+              and(
+                isElement,
+                and(
+                  hasNamespace(Namespace.HTML),
+                  not(hasRole(device, (role) => role.isWidget())),
+                  not(hasRole(device, "group")),
+                  not(isSemanticallyDisabled)
+                )
+              ),
+              node.parent().get()
+            ) &&
+            test(and(isText, isPerceivableForAll(device)), node)
+          ) {
+            textNodes = textNodes.append(node);
+          }
+
+          // Gather all aria-disabled widgets on the document
           if (
             test(
               and(
                 isElement,
                 or(
-                  not(hasNamespace(Namespace.HTML)),
                   hasRole(device, (role) => role.isWidget()),
                   and(hasRole(device, "group"), isSemanticallyDisabled)
                 )
               ),
               node
             )
           ) {
-            return;
+            // Compute the names of these elements
+            const names = node.attributes
+              .filter((attribute) => attribute.name === "name")
+              .map((attribute) => attribute.value);
+            disabledWidgetNames = disabledWidgetNames.concat(names);
```

Comment:
This should be about their accessible name, not the `name` attribute.
So, something like `ariaNode.from(node, device).name`.

The name does contain a `source` which recursively contains all nodes taking part in the computation: https://github.com/Siteimprove/alfa/blob/main/packages/alfa-aria/src/name.ts#L86-L91
(it may be worth to have the source implement `Iterable<dom.Node>` so they can each iterate the nodes taking part in their computation and we can just use this iterator here, this is done like that: https://github.com/Siteimprove/alfa/blob/main/packages/alfa-selector/src/selector.ts#L131-L133)